includes/class-wc-youcanpay-webhook-handler.php

```diff
@@ -38,30 +38,148 @@ public function __construct() {
 		add_action( 'woocommerce_api_wc_youcanpay', [ $this, 'check_for_webhook' ] );
 	}
 
-	/**
-	 * Check incoming requests for YouCan Pay Webhook data and process them.
-	 */
-	public function check_for_webhook() {
-		if ( ! array_key_exists( 'REQUEST_METHOD', $_SERVER )
-		     || ! array_key_exists( 'wc-api', $_GET )
-		     || ! array_key_exists( 'gateway', $_GET )
-		     || ( 'wc_youcanpay' !== $_GET['wc-api'] )
-		) {
-			return false;
-		}
-
-		switch ( wc_clean( wp_unslash( $_GET['gateway'] ) ) ) {
-			case WC_Gateway_YouCanPay::ID:
-				return $this->youcanpay_credit_card();
-			case WC_Gateway_YouCanPay_Standalone::ID:
-				return $this->youcanpay_standalone();
-		}
-
-		return false;
-	}
-
-	/**
-	 * @return bool
+    /**
+     * Check incoming requests for YouCan Pay Webhook data and process them.
+     */
+    public function check_for_webhook() {
+        if (!array_key_exists('REQUEST_METHOD', $_SERVER)
+            || !array_key_exists('wc-api', $_GET)
+            || ('wc_youcanpay' !== $_GET['wc-api'])
+        ) {
+            return false;
+        }
+
+        if (isset($_SERVER['REQUEST_METHOD']) && (strtoupper($_SERVER['REQUEST_METHOD']) === 'POST')) {
+            return $this->post_request();
+        } else {
+            return $this->get_request($_GET);
+        }
+    }
+
+    /**
+     * @param array $data
+     *
+     * @return bool
+     */
+    private function get_request($data) {
+        if (!array_key_exists('gateway', $data)) {
+            return false;
+        }
+
+        switch (wc_clean(wp_unslash($data['gateway']))) {
+            case WC_Gateway_YouCanPay::ID:
+                return $this->youcanpay_credit_card();
+            case WC_Gateway_YouCanPay_Standalone::ID:
+                return $this->youcanpay_standalone();
+        }
+
+        return false;
+    }
+
+    /**
+     * @return bool
+     * @throws WC_Data_Exception
+     */
+    private function post_request() {
+        $data = json_decode(file_get_contents('php://input'), true);
+
+        if (json_last_error() == JSON_ERROR_NONE) {
+            if (!array_key_exists('payload', $data)) {
+                return false;
+            }
+
+            if (!array_key_exists('transaction', $data['payload'])) {
+                return false;
+            }
+
+            if (!array_key_exists('payment_method', $data['payload'])) {
+                return false;
+            }
+
+            $transaction_id = null;
+            $transaction = new WC_YouCanPay_Transaction_Model($data['payload']['transaction']);
+            $payment_method = new WC_YouCanPay_Payment_Method_Model($data['payload']['payment_method']);
+            $payment_method_name = sprintf(__('YouCan Pay (%s)', 'youcan-pay'), $payment_method->get_name());
```

Comment:
Apply coding style by using camel case

Reply:
PHP Coding Standards for WordPress

Use lowercase letters in variable, action/filter, and function names (never camelCase). Separate words via underscores. Don’t abbreviate variable names unnecessarily; let the code be unambiguous and self-documenting.